tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/AppServiceTests.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.CodeDom;
+using System.Net;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.ManagedIdentity;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Test.Common;
+using Microsoft.Identity.Test.Common.Core.Mocks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.ManagedIdentityTests
+{
+    [TestClass]
+    public class AppServiceTests
+    {
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            TestCommon.ResetInternalStaticCaches();
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            SetEnvironmentVariables(null, null);
+        }
+
+        [DataTestMethod]
+        [DataRow("http://127.0.0.1:41564/msi/token/", "https://management.azure.com", "https://management.azure.com", null)]
```

Comment:
are we not testing the api-version?

Reply:
The api-version is hard coded, so it is not going to change. Do you mean if app service is used for a different api version? @gladjohn

Reply:
there are two cases, where api-version was not specified or a wrong api-version was specified, 

```
{ "statusCode": 400, "body": "{\"error\":{\"code\":\"ApiVersionUnspecified\",\"message\":\"An API version is required, but was not specified.\",\"innerError\":null}}" }
```

```
{ "statusCode": 400, "body": "{\"error\":{\"code\":\"UnsupportedApiVersion\",\"message\":\"The HTTP resource that matches the request URI 'http://127.0.0.1:41821/msi/token' does not support the API version '2018-08-01'.\",\"innerError\":null}}" }
``` 

since we hardcode the version the first case may not be applicable, but if the API-Version was changed in the resource? (if that could happen) then the second one may be applicable.

Reply:
This is same as getting an error response. I have added unit test for that scenario. I don't think we need to specifically add another unit test. Integration test for this would make more sense.